lib/Extension/LanguageServer/Listener/IncrementalUpdateListener.php

```diff
@@ -39,13 +40,6 @@ public function getListenersForEvent(object $event): Generator
             };
             return;
         }
-
-        if ($event instanceof TextDocumentIncrementallyUpdated) {
-            yield function (TextDocumentIncrementallyUpdated $updated): void {
-                $this->applyEdits($updated->identifier(), $updated->events());
-            };
-            return;
-        }
```

Comment:
this is currently handled by the AstUpdateListener -- and the question is if the AST listener should pro-actively update the AST (there's a possiblity it does so needlessly) or if we could buffer the edits for the AST - which would risk flooding it when it is requested - (in such a case we could have a buffer limit that when exceeded would trigger a single re-parse)...

another option may be to just make the incremental parser opt-in until it's sufficiently good enough that we can pro-actively parse.